bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/Shell.java

```diff
@@ -298,8 +300,28 @@ public Shell (Display display, int style) {
 	if (handle != 0 && !embedded) {
 		state |= FOREIGN_HANDLE;
 	}
+
+	int [] dpiX = new int [1];
+	int [] dpiY = new int [1];
+	int shellZoomFactor;
+	int shellNativeZoomFactor;
+	if (parent != null) {
+		shellZoomFactor = parent.getZoomFactor();
+		shellNativeZoomFactor = parent.getNativeZoomFactor();
+	} else {
+		Monitor monitor = getMonitor();
+		OS.GetDpiForMonitor (monitor.handle, 0, dpiX, dpiY);
+		int mappedDPIZoom = DPIUtil.mapDPIToZoom(dpiX[0]);
```

Comment:
This looks like exactly the implementation of `Display.getMonitor(long)`, so that I would expect `getMonitor().getZoom()` to return what is calculated as `mappedDPIZoom` here. Is there any difference?

Reply:
No, not anymore, you are correct. I replace it with getMonitor().getZoom()